index.bs

```diff
@@ -1878,13 +1877,32 @@ A {{WebTransportError}} has the following internal slots.
 
 <div algorithm>
 
+To <dfn for="WebTransportError">create</dfn> a {{WebTransportError}} error with a
+{{WebTransportErrorSource}} |source| and optionally a |streamErrorCode| defaulting
+to `null`, run these steps:
+
+1. Let |message| be an [=implementation-defined=] string.
+1. Let |options| be `{source, streamErrorCode}`.
+1. Return a new {{WebTransportError/constructor()}} with |message| and |options|.
+
+</div>
+
+<div algorithm>
+
 The <dfn constructor for="WebTransportError"
-lt="WebTransportError(init)">new WebTransportError(init)</dfn> constructor steps are:
+lt="WebTransportError(message, options)">new WebTransportError(message, options)</dfn>
+constructor steps are:
 
-1. Let |error| be [=this=].
-1. Let |message| be |init|.{{WebTransportErrorInit/message}} if it exists, and `""` otherwise.
-1. [=WebTransportError/Set up=] |error| with |message| and `"stream"`.
-1. Set |error|.{{WebTransportError/[[StreamErrorCode]]}} to |init|.{{WebTransportErrorInit/streamErrorCode}} if it exists.
+1. Set |this|'s internal slots as follows:
+    : {{WebTransportError/[[Source]]}}
+    :: |options|.{{WebTransportErrorOptions/source}}
+    : {{WebTransportError/[[StreamErrorCode]]}}
+    :: |options|.{{WebTransportErrorOptions/streamErrorCode}}
+1. Run the [=DOMException/new DOMException(message, name)=] constructor steps on
+   [=this=] with |message| and `"WebTransportError"`.
```

Comment:
This is not OK; follow the example at https://whatpr.org/webidl/1211.html#example-domexception-derived-interface .